File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        vector<int>results;
        int length=s.size();
        int g=length;
        int zeros=0;
        int subs=0;
        while(g && subs<length-1){
            int ones=0;
            if(s[subs]=='0'){
                zeros+=1;
            }
            for (int k=length-1; k>subs; k--){
                if(s[k]=='1'){
                    ones+=1;
                    }
            }        
            results.push_back(ones+zeros);
            subs+=1;
            g-=1;
        }
        return *max_element(results.begin(),results.end());

        
    }
};

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        l=[]
        length=len(s)
        g=length
        zeros=0
        subs=0
        while(g and subs<length-1):
            ones=0
            if(s[subs]=="0"):
                zeros+=1
            for k in range(length-1,subs,-1):
                if(s[k]=="1"):
                    ones+=1
            l.append(ones+zeros)
            subs+=1
            g-=1
        return max(l)

        